test/tests/functional/pbs_qrun.py

```diff
@@ -92,3 +94,44 @@ def test_invalid_host_val(self):
         msg = "Server is not up"
         self.assertTrue(self.server.isUp(), msg)
         self.logger.info("As expected server is up and running")
+
+    def test_qrun_hangs(self):
+        """
+        This test submit 500 jobs with differnt equivalence class,
+        turn of scheduling and qrun job to
+        verify whether qrun hangs.
+        """
+        node = self.mom.shortname
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1}, id=node)
+        for walltime in range(1, 501):
+            j = Job(TEST_USER)
+            a = {'Resource_List.walltime': walltime}
+            j.set_attributes(a)
+            if walltime == 500:
+                jid = self.server.submit(j)
+            else:
+                self.server.submit(j)
+        self.logger.info("Submitted 500 jobs with different walltime")
+
+        now = int(time.time())
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SCHED,
```

Comment:
Please set Expect to True here to make sure that scheduling is turned off before you proceed

Reply:
Sure.

Reply:
Here we are setting value to a SCHED object, if I use "expect=True" It will fail with below error.

2018-12-14 10:16:18,088 INFO     expect on server centosvm: scheduling set False && server_state set Idle sched attempt: 60
2018-12-14 10:16:18,594 INFO     FAILED
File "/usr/lib/python2.7/site-packages/ptl/lib/pbs_testlib.py", line 8112, in expect
    raise PtlExpectError(rc=1, rv=False, msg=_msg)
PtlExpectError: rc=1, rv=False, msg=expected on server centosvm: scheduling set False && server_state set Idle sched attempt: 60

Reply:
Sure.